numerical/area/bound.py

```diff
@@ -0,0 +1,137 @@
+import numpy as np
+from abc import ABCMeta, abstractmethod
+import matplotlib.pyplot as plt
+
+
+class _BaseBoundary(metaclass=ABCMeta):
+    @abstractmethod
+    def plot(self, save_path=None):
+        """ Visualizes boundary. """
+
+    def _set_described_rect(self):
+        """ Set n-dim rectangular bound which describes initial bound. """
+
+
+class _BoundaryIterable(metaclass=ABCMeta):
+    """ The class represents iterator interface for boundaries. """
+    def __init__(self, data: list):
+        self._data = data
+        self._bounds_count = len(data)
+
+    def __iter__(self):
+        self._bound_id = -1
```

Comment:
why `-1`?

Reply:
to avoid 
```return self._data[self._bound_id  - 1]```

Reply:
Where?
I believe the proper initial value is `0`